src/test/java/seedu/address/logic/parser/IndexIdPairTest.java

```diff
@@ -0,0 +1,313 @@
+        // person Id only
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        assertTrue(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_PERSON_ID));
+
+        // location Id only
+        argMultimap = ArgumentTokenizer.tokenize(locationIdString, PREFIX_LOCATION_ID);
+        assertTrue(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+    }
+
+    @Test
+    public void checkIndexOrIdOnly_indexAndIdOrNeither_false() {
+        // person Id and person Index
+        argMultimap = ArgumentTokenizer.tokenize(indexString + personIdString, PREFIX_PERSON_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_PERSON_ID));
+
+        // location Id and location Index
+        argMultimap = ArgumentTokenizer.tokenize(indexString + locationIdString, PREFIX_LOCATION_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+
+        // empty string
+        argMultimap = ArgumentTokenizer.tokenize(empty, PREFIX_LOCATION_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+    }
+
+    @Test
+    public void checkIndexOrIdOnly_wrongPrefix_false() {
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+
+        argMultimap = ArgumentTokenizer.tokenize(locationIdString, PREFIX_LOCATION_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_PERSON_ID));
+    }
+
+    @Test
+    public void constructor_nullValues_throwNullPointerException() {
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        assertThrows(NullPointerException.class, () -> new IndexIdPair(null, null));
+        assertThrows(NullPointerException.class, () -> new IndexIdPair(null, PREFIX_PERSON_ID));
+        assertThrows(NullPointerException.class, () -> new IndexIdPair(argMultimap, null));
+    }
+
+    @Test
+    public void constructor_validArgs_success() {
+        // person Id
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        IndexIdPair expectedPersonPair = new IndexIdPair(null, personId, PREFIX_PERSON_ID);
+        try {
+            assertEquals(new IndexIdPair(argMultimap, PREFIX_PERSON_ID), expectedPersonPair);
+        } catch (ParseException pe) {
+            throw new IllegalArgumentException("Invalid userInput.", pe);
+        }
+
+        // location index
+        argMultimap = ArgumentTokenizer.tokenize(indexString, PREFIX_LOCATION_ID);
+        IndexIdPair expectedLocationPair = new IndexIdPair(INDEX_FIRST, null, PREFIX_LOCATION_ID);
+        try {
+            assertEquals(new IndexIdPair(argMultimap, PREFIX_LOCATION_ID), expectedLocationPair);
+        } catch (ParseException pe) {
+            throw new IllegalArgumentException("Invalid userInput.", pe);
+        }
```

Comment:
Are the try-catch statements required here as the test is not supposed to fail anyway, and the parse exception isn't part of an assertion.

Reply:
indexIdPair constructor might throw a ParseException. Hence, the solution is either to use a try catch block, or have the test method throw the parse exception. I chose to use try catch in this case.

Reply:
Ok, i went to look through other tests and it seems addressbook for person tests chooses to have the method throw exception. I will follow accordingly.